packages/contract/src/contract-factory.ts

```diff
@@ -146,26 +155,190 @@ export default class ContractFactory {
+        blobId: transactionRequest.blobId,
+      };
+    });
+
+    // Check the account can afford to deploy all chunks
+    let totalCost = bn(0);
+    const chainInfo = account.provider.getChain();
+    const gasPrice = await account.provider.estimateGasPrice(10);
+    const estimatedBlobIds: string[] = [];
+
+    for (const { transactionRequest, blobId } of chunks) {
+      if (!estimatedBlobIds.includes(blobId)) {
+        const minGas = transactionRequest.calculateMinGas(chainInfo);
+        const minFee = calculateGasFee({
+          gasPrice,
+          gas: minGas,
+          priceFactor: chainInfo.consensusParameters.feeParameters.gasPriceFactor,
+          tip: transactionRequest.tip,
+        }).add(1);
+
+        totalCost = totalCost.add(minFee);
+        estimatedBlobIds.push(blobId);
+      }
+    }
+    if (totalCost.gt(await account.getBalance())) {
+      throw new FuelError(ErrorCode.FUNDS_TOO_LOW, 'Insufficient balance to deploy contract.');
+    }
+
+    // Upload the blob if it hasn't been uploaded yet. Duplicate blob IDs will fail gracefully.
+    const uploadedBlobs: string[] = [];
+
+    // Deploy the chunks as blob txs
+    for (const { blobId, transactionRequest } of chunks) {
+      if (!uploadedBlobs.includes(blobId)) {
+        const fundedBlobRequest = await this.fundTransactionRequest(
+          transactionRequest,
+          deployOptions
+        );
+
+        let result: TransactionResult<TransactionType.Blob>;
+
+        try {
+          const blobTx = await account.sendTransaction(fundedBlobRequest);
+          result = await blobTx.waitForResult();
+        } catch (err: unknown) {
+          // Core will throw for blobs that have already been uploaded, but the blobId
+          // is still valid so we can use this for the loader contract
+          if ((<Error>err).message.indexOf(`BlobId is already taken ${blobId}`) > -1) {
+            // eslint-disable-next-line no-continue
+            continue;
+          }
+
+          throw new FuelError(ErrorCode.TRANSACTION_FAILED, 'Failed to deploy contract chunk');
+        }
+
+        if (!result.status || result.status !== TransactionStatus.success) {
+          throw new FuelError(ErrorCode.TRANSACTION_FAILED, 'Failed to deploy contract chunk');
+        }
+
+        uploadedBlobs.push(blobId);
+      }
+    }
+
+    // Get the loader bytecode
+    const blobIds = chunks.map(({ blobId }) => blobId);
+    const loaderBytecode = getLoaderInstructions(blobIds);
+
+    // Deploy the loader contract via create tx
+    const { contractId, transactionRequest: createRequest } = this.createTransactionRequest({
+      bytecode: loaderBytecode,
+      ...deployOptions,
+    });
+    await this.fundTransactionRequest(createRequest, deployOptions);
+    const transactionResponse = await account.sendTransaction(createRequest);
 
     const waitForResult = async () => {
       const transactionResult = await transactionResponse.waitForResult<TransactionType.Create>();
```

Comment:
Are you going to be putting the sending of blob transactions + loader contract into `waitForResult` as we discussed? 
For others, the idea is that deploying a large contract should behave the same as deploying a small contract in the sense that the end user shouldn't be blocked by transaction execution. For small contracts, we deploy the contract in a single transaction which the user can await with `waitForResult`. Based on the current implementation for large contracts, the user would be blocked by the sending and execution of all blob transactions and their call of `waitForResult` would be just on the last loader contract transaction. Ideally the `waitForResult` would also encompass the blob transactions as well so that, from the perspective of the user, it all behaves as a single transaction.

Reply:
I've already added it as a follow point to #2917. I took a brief look the other day but had an issue with the loader contract `sendTransaction` call. Will try get some more time today.

Reply:
@nedsalk Good catch. This is a DX must follow up with the rest.

@danielbate If we can, we should ship this DX change with this PR.

Reply:
Given testnet is ready I'm going to do it as follow up, I've got a PR but it's not perfect and requires some discussion.

Reply:
Would this require introducing new breaking changes later?

Reply:
@arboleya It doesn't seem like it would require breaking changes because we'd just put the logic inside of the `waitForResult` that's already returned from the method.

Reply:
My current problem is with returning the transaction ID for the create tx. Putting everything inside `waitForResult` means that we don't that ID until we've completed all the blob TXs and then funded the create TX. I should have some time this afternoon. You're welcome to take a look if you have time @nedsalk. I've got a draft in #2929.

Reply:
@danielbate I took a look and, from what it looks like, this is a problem we won't be able to solve without partially (or fully?) implementing #2678. We'd have to know upfront which UTXOs the blob TXs will spend and the new UTXOs they'll create up until the last blob TX's UTXOs which we'd then use for the create TX - something like that. Right now the only thing we can do is document for the users that deploying contracts as blobs might hang a bit longer on the `await`.

Reply:
I'd considered making transaction ID a promise but obviously a breaking change. For now I'll add additional docs. Thanks for taking a look.